internal/storage/v2/grpc/trace_storage.proto

```diff
@@ -83,4 +129,27 @@ service TraceReader {
   // GetOperations returns all operation names for a given service
   // known to the backend from traces within its retention period.
   rpc GetOperations(GetOperationsRequest) returns (GetOperationsResponse) {}
+
+  // FindTraces returns a stream that retrieves traces matching query parameters.
+  //
+  // The chunking rules are the same as for GetTraces.
+  //
+  // If no matching traces are found, an empty stream is returned.
+  // If an error is encountered, the iterator returns the error and stops.
+  //
+  // There's currently an implementation-dependent ambiguity whether all query filters
+  // (such as multiple tags) must apply to the same span within a trace, or can be satisfied
+  // by different spans.
+  rpc FindTraces(FindTracesRequest) returns (stream opentelemetry.proto.trace.v1.TracesData) {}
+
+  // FindTraceIDs returns a stream that retrieves IDs of traces matching query parameters.
+  //
+  // If no matching traces are found, an empty stream is returned.
+  // If an error is encountered, the stream returns the error and stops.
+  //
+  // This call behaves identically to FindTraces, except that it returns only the list
+  // of matching trace IDs. This is useful in some contexts, such as batch jobs, where a
+  // large list of trace IDs may be queried first and then the full traces are loaded
+  // in batches.
+  rpc FindTraceIDs(FindTracesRequest) returns (stream bytes) {}
```

Comment:
@yurishkuro is it fine to use the same request object for both or do we want different wrappers?

Reply:
I think it's fine